Events/MessageEvent.cs

```diff
@@ -119,11 +119,37 @@ public static async Task MessageHandlerAsync(DiscordClient client, DiscordMessag
                         return;
                     }
 
+                    DiscordMessageBuilder memberWarnInfo = new();
+
                     DiscordRole muted = message.Channel.Guild.GetRole(Program.cfgjson.MutedRole);
                     if (modmailMember.Roles.Contains(muted))
                     {
-                        await channel.SendMessageAsync(new DiscordMessageBuilder().AddEmbed(await WarningHelpers.GenerateWarningsEmbedAsync(modmailMember)).AddEmbed(await MuteHelpers.MuteStatusEmbed(modmailMember, message.Channel.Guild)));
+                        memberWarnInfo.AddEmbed(await WarningHelpers.GenerateWarningsEmbedAsync(modmailMember)).AddEmbed(await MuteHelpers.MuteStatusEmbed(modmailMember, message.Channel.Guild));
                     }
+                    
+                    // Add notes to message if any exist & are set to show on modmail
+                    
+                    // Get user notes
+                    var notes = Program.db.HashGetAll(modmailMember.Id.ToString())
+                        .Where(x => JsonConvert.DeserializeObject<UserNote>(x.Value).Type == WarningType.Note).ToDictionary(
+                            x => x.Name.ToString(),
+                            x => JsonConvert.DeserializeObject<UserNote>(x.Value)
+                        );
+                    
+                    // Filter to notes set to notify on modmail
+                    var notesToNotify = notes.Where(x => x.Value.ShowOnModmail).ToDictionary(x => x.Key, x => x.Value);
+                    
+                    // If there are notes, build embed and add to message
+                    if (notesToNotify.Count != 0)
+                        memberWarnInfo.AddEmbed(await UserNoteHelpers.GenerateUserNotesEmbedAsync(modmailMember, notesToUse: notesToNotify));
+                    
+                    // If message was built (if user is muted OR if user has notes to show on modmail), send it
+                    if (memberWarnInfo.Embeds.Count != 0) // todo: this is probably not the best way to check this?
```

Comment:
> this is probably not the best way to check this?

I think this actually makes perfectly sense to me.

Reply:
Okay! Will leave it as-is then.